FlowCrypt/Controllers/SetupImap/SetupImapViewController.swift

```diff
@@ -486,16 +486,20 @@ extension SetupImapViewController {
     private func connect() {
         view.endEditing(true)
 
-        let result = checkCurrentUser()
-        switch result {
-        case .failure(.empty):
-            showToast("other_provider_error_other".localized)
-        case .failure(.password):
-            showToast("other_provider_error_password".localized)
-        case .failure(.email):
-            showToast("other_provider_error_email".localized)
-        case .success:
+        do {
+            try checkCurrentUser()
             checkImapSession()
+        } catch {
+            switch error as? UserError {
+            case .empty:
+                showToast("other_provider_error_other".localized)
+            case .password:
+                showToast("other_provider_error_password".localized)
+            case .email:
```

Comment:
Interesting that we don't throw this error

Reply:
Yes, this error handling possibly needs reworking in general. There is no handling of other errors, also.

Reply:
But for now never mind, IMAP/SMTP is low priority atm.